src/Analyzers/Core/CodeFixes/NamingStyle/NamingStyleCodeFixProvider.cs

```diff
@@ -163,12 +163,17 @@ protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperation
 #if CODE_STYLE  // https://github.com/dotnet/roslyn/issues/42218 tracks removing this conditional code.
                 return SpecializedCollections.SingletonEnumerable(codeAction);
 #else
-                var factory = _startingSolution.Services.GetRequiredService<ISymbolRenamedCodeActionOperationFactoryWorkspaceService>();
-                return new CodeActionOperation[]
+
+                using var _ = PooledObjects.ArrayBuilder<CodeActionOperation>.GetInstance(out var operations);
```

Comment:
```suggestion
                using var operations = TemporaryArray<CodeActionOperation>.Empty;
```

Reply:
❓ what's the difference between the two?

Reply:
I believe it is design to held small amount of objects.

Reply:
Yup. For when you know you have 4 or fewer elements, it's more efficient.